src/app/src/components/UserApiInfo.jsx

```diff
@@ -0,0 +1,165 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { arrayOf, func, string, bool } from 'prop-types';
+
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import IconButton from '@material-ui/core/IconButton';
+import InfoIcon from '@material-ui/icons/Info';
+import Tooltip from '@material-ui/core/Tooltip';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+import {
+    userApiInfoTooltipTitles,
+    IS_NOT_SET,
+    FORBIDDEN,
+} from '../util/constants';
+
+import { fetchUserApiInfo } from '../actions/profile';
+import { userApiInfoPropType } from '../util/propTypes';
+
+const componentStyles = Object.freeze({
+    withoutMargin: Object.freeze({
+        marginBlockStart: '0',
+        marginBlockEnd: '0',
+    }),
+});
+
+const StyledTooltip = withStyles({
+    tooltip: {
+        color: 'rgba(0, 0, 0, 0.8)',
+        fontSize: '0.875rem',
+        backgroundColor: 'white',
+        border: 'solid rgba(0, 0, 0, 0.25)',
+        borderRadius: '10px',
+        padding: '10px',
+        lineHeight: '1',
+    },
+})(Tooltip);
+
+class UserAPIInfo extends Component {
```

Comment:
Please, rewrite it to the functional component.
See the recommendation about functional components in [the React docs](https://react.dev/reference/react/Component#alternatives).
<img width="1166" alt="Screenshot 2024-02-19 at 12 37 53" src="https://github.com/opensupplyhub/open-supply-hub/assets/58369841/6f025b80-fe61-4244-8386-ace8facdb0eb">

Reply:
rewritten